elastic/kibana
# [Uptime] Delete memory and domain classes

## Summary

Resolves https://github.com/elastic/uptime/issues/85.

When I created the Uptime plugin, I copied a significant amount of boilerplate from some other newer Kibana plugins. As time has gone on and myself and my teammates have maintained the code for about a year, we've discovered that certain test mechanisms designed by others have been less useful and more time consuming for us.

In this patch, I've deleted a number of files and removed a layer of abstraction. I have found this level of abstraction hinders fast development and requires a lot of tinkering, it's also non-obvious to newcomers to the codebase. Most of the type of testing this deleted code allows for can be adequately simulated with mocks, and the benefit of allowing the plugin to run decoupled from the environment for which it's designed is outweighed by the overhead of maintaining separate class implementations for us at this time.

## Testing
As long as CI is passing, you should be able to load up this patch and smoke test it without issues. It should function exactly as it would if you were running `master`.

0. Start up a stack running this patch (Heartbeat, ES, Kibana)
1. Navigate to Uptime. Ensure you're seeing monitor data like you'd expect
2. Drill into one or more monitors, ensure the data on the page looks correct
